gl/gloffscreen_glx.c

```diff
@@ -95,17 +95,51 @@ GloContext *glo_context_create(int formatFlags)
 #endif
 
     /* Create GLX context */
-    context->glx_context = glXCreateNewContext(x_display, configs[0], GLX_RGBA_TYPE, NULL, True);
+
+    /* FIXME: Check for "GLX_ARB_create_context" extension*/
+    #define GLX_CONTEXT_MAJOR_VERSION_ARB     0x2091
+    #define GLX_CONTEXT_MINOR_VERSION_ARB     0x2092
+    #define GLX_CONTEXT_PROFILE_MASK_ARB     0x9126
+    #define GLX_CONTEXT_CORE_PROFILE_BIT_ARB 0x00000001
+    typedef GLXContext (*GLXCREATECONTEXTATTRIBSARBPROC)(Display*, GLXFBConfig, GLXContext, Bool, const int*);
+    GLXCREATECONTEXTATTRIBSARBPROC glXCreateContextAttribsARB = (GLXCREATECONTEXTATTRIBSARBPROC) glXGetProcAddress((const GLubyte*)"glXCreateContextAttribsARB");
+    if (glXCreateContextAttribsARB == NULL) return NULL;
+    int context_attribute_list[] = {
+        GLX_CONTEXT_MAJOR_VERSION_ARB, 3,
+        GLX_CONTEXT_MINOR_VERSION_ARB, 3,
+        GLX_CONTEXT_PROFILE_MASK_ARB, GLX_CONTEXT_CORE_PROFILE_BIT_ARB,
+        None
+    };
+    context->glx_context = glXCreateContextAttribsARB(x_display, configs[0], 0, True, context_attribute_list);
+    if (context->glx_context == NULL) {
+      assert(0);
```

Comment:
oops, that's actually just a debug check when things went a little wrong.
Fails gracefully (by returning NULL - not sure how that is handled) a check later.